fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/impl/transfer/fluid/CombinedProvidersImpl.java

```diff
@@ -83,8 +83,12 @@ public Storage<FluidVariant> find(ItemStack itemStack, ContainerItemContext cont
 
 	public static Event<FluidStorage.CombinedItemApiProvider> getOrCreateItemEvent(Item item) {
 		// register here is thread-safe, so the query below will return a valid provider (possibly one registered before or from another thread).
-		FluidStorage.ITEM.registerForItems(new Provider(), item);
 		ItemApiLookup.ItemApiProvider<Storage<FluidVariant>, ContainerItemContext> existingProvider = FluidStorage.ITEM.getProvider(item);
+		if (existingProvider == null) {
+			// if you always register, lookup will spam warning, so only register when not registered
+			existingProvider = new Provider();
+			FluidStorage.ITEM.registerForItems(existingProvider, item);
+		}
```

Comment:
Dont make unrelated changes like this, if needed make another PR

Reply:
I didn't know before that changes from the same branch would be automatically merged into the same PR. My original intent was opening another PR. Now maybe I should roll it back.